zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java

```diff
@@ -283,6 +342,154 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
 
                 removeHistoryReferenceAlerts(rootNode);
             }
+        } else if (ACTION_UPDATE_ALERT.equals(name)) {
+            int alertId = ApiUtils.getIntParam(params, PARAM_ALERT_ID);
+            String alertName = ApiUtils.getNonEmptyStringParam(params, PARAM_ALERT_NAME);
+            int riskId = ApiUtils.getIntParam(params, PARAM_ALERT_RISK);
+            int confidenceId = ApiUtils.getIntParam(params, PARAM_ALERT_CONFIDENCE);
+            String desc = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_DESCRIPTION);
```

Comment:
This is misleading, the parameter is mandatory. Was it meant to be optional?

Reply:
No just didn’t see another obvious String method.

Reply:
Your earlier comment addresses this, dunno why I didn’t see that autocomplete option I. Eclipse

Reply:
OK, for param/attack/evidence/other info/refs might be better if optional?

Reply:
Sure

Reply:
CWE/WASC IDs too.